statistics/mode.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns the arithmetic mode(s) of the numbers in the given collection
+ *
+ * Example:
+ *
+ * ```ts
+ * import { assertEquals } from "../testing/asserts.ts"
+ * import { mode } from "./mode.ts"
+ *
+ * const numbers = [ 4, 2, 7, 4 ]
+ * assertEquals(mode(numbers), 4)
+ * ```
+ */
+export function mode(collection: Array<number>): Set<number> | undefined {
+  if (collection.length === 0) {
+    return undefined;
+  }
+
+  const counter: Record<number, number> = {};
+  counter[collection[0]] = 1;
+  const maxes = new Set([collection[0]]);
+  let maxCount = 1;
+
+  for (let i = 1; i < collection.length; i++) {
+    const number = collection[i];
+    counter[number] = (counter[number] || 0) + 1;
+
+    if (counter[number] === maxCount) {
+      maxes.add(number);
+    } else if (counter[number] > maxCount) {
```

Comment:
I would try to reduce the calls to `counter[number]` to exactly one read and exactly one write, for performance reasons. You should be able to just read it into a variable, and reference that in the block below.